src/main/java/com/epam/test_generator/controllers/CaseController.java

```diff
@@ -171,4 +164,10 @@
 
         return new ResponseEntity<>(HttpStatus.OK);
     }
+
+    @RequestMapping(value = "/suits/{suidId}/cases/{caseId}/history", method = RequestMethod.GET)
+    public ResponseEntity<List<CaseVersionDTO>> getCaseHistory(@PathVariable("suidId") long suitId,
+                                                               @PathVariable("caseId") long caseId) {
+        return new ResponseEntity<>(caseService.getCaseHistory(suitId, caseId), HttpStatus.OK);
+    }
```

Comment:
Please, investigate and implement way to rollback entity to given historical point. Expose method in controller to perform the rollback.